src/nspawn/nspawn-stub-pid1.c

```
@@ -55,18 +55,21 @@ int stub_pid1(sd_id128_t uuid) {
 
         /* Surrender the terminal this stub may control so that child processes can have a controlling terminal
          * without resorting to setsid hacks. */
-        r = ioctl(STDIN_FILENO, TIOCNOTTY);
-        if (r < 0 && errno != ENOTTY)
-                return log_error_errno(errno, "Failed to surrender controlling terminal: %m");
-
+        if (arg_console_mode != CONSOLE_PIPE) {
+                r = ioctl(STDIN_FILENO, TIOCNOTTY);
+                if (r < 0 && errno != ENOTTY)
+                        return log_error_errno(errno, "Failed to surrender controlling terminal: %m");
+        }
         pid = fork();
         if (pid < 0)
                 return log_error_errno(errno, "Failed to fork child pid: %m");
 
         if (pid == 0) {
                 /* Return in the child */
                 assert_se(sigprocmask(SIG_SETMASK, &oldmask, NULL) >= 0);
-                setsid();
+                if (arg_console_mode != CONSOLE_PIPE) {
+                        setsid();
+                }
```

No {} around single-line if blocks, please, see https://systemd.io/CODING_STYLE/#formatting

Hmm, given that you touch this line here, please prefix the `setsid()` with a cast to `(void)` so that static checkers understand that we knowingly ignore errors here

---

Fixed both, thanks!